### PR TITLE
Lead member

### PR DESCRIPTION
**Pull request by maximenkom**
Modified sendMail method in userServiceImpl, created mockito testing for allianceService, userService and allianceController

### CODE REVIEW

File: src/main/java/ua/cv/tim/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -148,14 +150,14 @@ public void addUser(UserDTO member) throws MessagingException {
 		player.setAlliance(allianceDao.getAllianceByName(member.getAlliance()));
 		user.setPlayer(player);
 		playerDao.add(player);
-		sendEmail(user, password);
+		sendEmail(user,"Your login is \"" + user.getLogin() + "\", and password: \""
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/service/impl/UserServiceImpl.java:153]: I think, the second parameter here is redundant. You can create this message in sendEmail() method

File: src/test/java/ua/cv/tim/service/impl/AllianceServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    public void setUp(){
+        MockitoAnnotations.initMocks(this);
+        alliancesList= getAlliancesList();
+    }
+    @AfterMethod
+    public void resetAllMocks(){
+        Mockito.reset(allianceDao,userService,userDao,sendMail);
+    }
+    @Test
+    public void getAllTest(){
```
Comment (ViacheslavKruhlov) [inline src/test/java/ua/cv/tim/service/impl/AllianceServiceImplTest.java:66]: ununderstandable test name, try to write more detail methods names

File: src/test/java/ua/cv/tim/service/impl/AllianceServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        List<AllianceDTO> allianceDTOsList = new ArrayList<>();
+        for(Alliance alliance:alliancesList){
+            allianceDTOsList.add(convertAllianceToDTO(alliance));
+        }
+        when(allianceDao.getAll()).thenReturn(alliancesList);
+        assertEquals(allianceService.getAll(),allianceDTOsList);
+        verify(allianceDao, times(1)).getAll();
```
Comment (VyacheslavKruhlov) [inline src/test/java/ua/cv/tim/service/impl/AllianceServiceImplTest.java]: after that verify, need to check verifyNoMoreInteractions(), in all tests

File: src/test/java/ua/cv/tim/service/impl/UserServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        MockitoAnnotations.initMocks(this);
+        users=getUserList();
+    }
+    @AfterMethod
+    public void resetAllMocks(){
+        Mockito.reset(allianceDao,playerDao,userDao,sendMail);
+    }
+
+    @Test
+    public void getUserByUsernameTest() {
```
Comment (ViacheslavKruhlov) [inline src/test/java/ua/cv/tim/service/impl/UserServiceImplTest.java:60]: naming convention - word 'test' in the begining of method name - testSomeLogic()

**Review by RomaMocherniuk — APPROVED** [review]

File: src/main/java/ua/cv/tim/service/impl/UserServiceImpl.java (hunk 1 above)
Comment (maximenkom) [reply src/main/java/ua/cv/tim/service/impl/UserServiceImpl.java:153]: message created inside method before modification, but now we need to modify message for create and update methods, so  decision to modify message text in params was taken and method was modified

File: src/test/java/ua/cv/tim/service/impl/AllianceServiceImplTest.java (hunk 2 above)
Comment (maximenkom) [reply src/test/java/ua/cv/tim/service/impl/AllianceServiceImplTest.java:66]: test name made from real method name + suffix test, and method getAll from allianceServiceImpl class returns all alliances appropriately

File: src/test/java/ua/cv/tim/service/impl/UserServiceImplTest.java (hunk 4 above)
Comment (maximenkom) [reply src/test/java/ua/cv/tim/service/impl/UserServiceImplTest.java:60]: ok, I'll redefine methods

**Review by olleg12 — APPROVED** [review]

**Review by ViacheslavKruhlov — APPROVED** [review]

**Pull request merged**